client/js/controllers/user/profile.view.js

```diff
@@ -0,0 +1,18 @@
+angular
+  .module('app')
+  .controller('ProfileViewController', ['$scope', '$state', 'AuthService', 'Affiliation', 'User', '$stateParams', function ($scope, $state, AuthService, Affiliation, User, $stateParams) {
+
+    User.findById({id: $stateParams.userId, include: ['useraffiliation']}).$promise.then(function(u) {
```

Comment:
Ja der call ist jetzt korrekt. Aber wie gesagt, die n:m relation scheint veraltet. Wenn ich im Profil meine affiliation ändere dann wird die affiliationId direkt mit dem user verknüpft anstatt es in die Zwischentabelle einzutragen. 
D.h. der fix der relation im user model fehlt noch.